### PR TITLE
Add hackfmt fixer

### PR DESCRIPTION
**Pull request by SamHowie**
Adding hackfmt fixer for Hack

### CODE REVIEW

**Comment by mitermayer** [comment]
Awesome!!!

File: autoload/ale/fixers/hackfmt.vim [hunk 1]
```diff
@@ -0,0 +1,17 @@
+" Author: Sam Howie <samhowie@gmail.com>
+" Description: Integration of hackfmt with ALE.
+
+call ale#Set('php_hackfmt_executable', 'hackfmt')
+call ale#Set('php_hackfmt_options', '-i')
```
Comment (w0rp) [inline autoload/ale/fixers/hackfmt.vim]: What is the meaning of `-i`? If it's essential to running the program, you might leave that out of configuration.

File: test/fixers/test_hackfmt_fixer_callback.vader [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+
+Execute(The hackfmt callback should include custom hackfmt options):
+  let g:ale_php_hackfmt_options = "--skip-children"
+  call ale#test#SetFilename('../hack_files/testfile.php')
+
+  AssertEqual
+  \ {
+  \   'read_temporary_file': 1,
+  \   'command': ale#Escape('xxxinvalid')
+  \     . ' ' . g:ale_php_hackfmt_options
```
Comment (w0rp) [inline test/fixers/test_hackfmt_fixer_callback.vader]: Write a string literal here instead, rather than repeating the variable.

**Comment by fredemmott** [comment]
Would it be practical to use `git diff | hackfmt --diff` instead when in a git repository, + the hg equivalent? This will make hackfmt only run on the modified lines rather than the whole file.

Otherwise, `-i`, is required - it's basically equivalent to the `-i` flag in sed

File: autoload/ale/fixers/hackfmt.vim (hunk 1 above)
Comment (SamHowie) [reply autoload/ale/fixers/hackfmt.vim]: `-i` id the in-place option. It tells hackfmt to format the file in place.

It is essential, so I will move it out of the configuration.

File: test/fixers/test_hackfmt_fixer_callback.vader (hunk 2 above)
Comment (SamHowie) [reply test/fixers/test_hackfmt_fixer_callback.vader]: Will do!

**Comment by SamHowie** [comment]
@fredemmott, I like your idea!

I looked into implementing it and ran into a few difficulties which I need to figure out.

I propose that we keep this pull request simple and add the `hackfmt --diff` optimization afterwards.

**Pull request merged**